### DISCUSSION TITLE
Make taking payment user friendly

### DISCUSSION THREAD

**Discussion by lukepuplett** [WalletConnect/walletconnect-monorepo#3272] [Developer Support]
Hello everyone, I would like to accept a crypto payment on my website for the first time. I know that I can present a QR code, but I’m expecting customers to have a Wallet browser extension. How can I invoke their wallet browser extension?

For example, Solana have invented a URL scheme, which they call Solana Pay. When a suitable wallet app is installed on the user’s machine or device it will recognise such URLs and deep link into a new payment user interface. The new payment is pre-filled with the amount and the address and a note. I used this URL once to make a really cool NFC tip jar! True story.

Would WalletConnect, when implemented on my website, make it a more seamless experience to make a payment?

Imagine an invoice on the webpage with a QR code, and a Blockchain address that they can manually copy, is there a way I can add a button which when clicked invokes the browser extension wallet with the Blockchain address prefilled?

Even if not, WalletConnect looks like a really neat project anyway.

Thanks peeps!

**Comment by lukepuplett** [comment]
I've been reading and even trying to use AI to understand what WalletConnect and/or Web3Modal do, and how to use them.

The WalletConnect website is confusing. I can't tell who the website is talking to. App developers? Wallet developers? Dapp and wallet users? It's too salesy like it's marketing a B2B product. Lovely graphics but _what does it do_?

The docs help me to understand how to integrate the modal into my website's HTML (I'm using an old skool multi-page website).

But then what? It doesn't explain what the point of it is. By following your docs I can see how I might end-up with a button and a modal widget that links my website (dapp?*) and a user's wallet in some way, and then nothing. Why doesn't it say what I can do once it's connected? How does it help me to, e.g. submit a transaction to a contract, or sign some EIP 712 data or use it with ethers.js.

*my website is not a dapp; I have no code deployed to any blockchain, it's just a website that I want to allow people to pay for things in crypto.

**Reply by Soolan** [reply]
I 2nd this, and would love to know how to use WalletConnect and prefill tx details on a webpage (not a dApp)?
